diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/CompactInitialization.kt

```diff
@@ -145,7 +147,24 @@ class CompactInitialization(configRules: List<RulesConfig>) : DiktatRule(
         ?: run {
             // add apply block
             property.node.run {
-                val newInitializerNode = kotlinParser.createNode("${property.initializer!!.text}.apply {}")
+                val isNotRequiresParentheses = property.initializer.let {
+                    // list of expression types, that can be directly followed by a dot-qualified call
+                    // e.g. val x = foo()  ->  val x = foo().apply {}
+                    // e.g. val x = foo + bar  ->  val x = (foo + bar).apply {}
+                    it is KtParenthesizedExpression || it is KtDotQualifiedExpression ||
+                            it is KtCallExpression || it is KtReferenceExpression
+                }
+                val newInitializerNodeText = buildString {
+                    if (!isNotRequiresParentheses) {
+                        append("(")
+                    }
+                    append(property.initializer!!.text)
+                    if (!isNotRequiresParentheses) {
+                        append(")")
+                    }
+                    append(".apply {}")
```

Comment:
I'm sorry, but I didn't get why we are adding empty apply block here...

Reply:
We are filling it with statements later. This creation is to have a proper AST representation of `apply` block